browser/javascript/v2/react-app/src/ampli/index.js

```diff
@@ -0,0 +1,566 @@
+    this.event_type = 'Event With Optional Properties';
+    this.event_properties = properties;
+  }
+}
+
+export class EventWithTemplateProperties {
+  constructor(properties) {
+    this.event_type = 'Event With Template Properties';
+    this.event_properties = properties;
+  }
+}
+
+// prettier-ignore
+export class Ampli {
+  constructor() {
+    /* @type {AmplitudeClient|undefined} */
+    this.amplitude = undefined;
+    this.disabled = false;
+  }
+
+  /**
+   * @return {AmplitudeClient}
+   */
+  get client() {
+    this.isInitializedAndEnabled();
+    return this.amplitude;
+  }
+
+  /**
+   * @return {boolean}
+   */
+  get isLoaded() {
+    return this.amplitude != null;
+  }
+
+  /**
+   * @private
+   * @return {boolean}
+   */
+  isInitializedAndEnabled() {
+    if (!this.isLoaded) {
+      console.error('ERROR: Ampli is not yet initialized. Have you called ampli.load() on app start?');
+      return false;
+    }
+    return !this.disabled;
+  }
+
+  /**
+   * Initialize the Ampli SDK. Call once when your application starts.
+   * @param {LoadOptions} [options] Configuration options to initialize the Ampli SDK with.
+   */
+  load(options) {
+    this.disabled = options?.disabled ?? false;
+
+    if (this.isLoaded) {
+      console.warn('WARNING: Ampli is already initialized. Ampli.load() should be called once at application startup.');
+      return;
+    }
+
+    const env = options?.environment ?? 'development';
+    const apiKey = options?.client?.apiKey ?? ApiKey[env];
+
+    if (options?.client?.instance) {
+      this.amplitude = options?.client?.instance;
+    } else if (apiKey) {
+      this.amplitude = amplitude;
+      return this.amplitude?.init(apiKey, undefined, { ...DefaultOptions, ...options?.client?.options });
+    } else {
+      console.error("ERROR: ampli.load() requires 'environment', 'client.apiKey', or 'client.instance'");
+    }
+  }
+
+  /**
+   * Identify a user and set or update that user's properties.
+   *
+   * @param {string|undefined} userId The user's id.
+   * @param {Object} properties The user's properties.
+   * @param {string[]} [properties.optionalArray] Description for identify optionalArray
+   * @param {number} properties.requiredNumber Description for identify requiredNumber
+   * @param {IdentifyOptions} [options] Optional event options.
+   */
+  identify(userId, properties, options) {
+    if (!this.isInitializedAndEnabled()) {
+      return;
+    }
+
+    if (userId) {
+      options = Object.assign({}, options, { user_id: userId });
+    }
+
+    const ampIdentify = new amplitude.Identify();
+    if (properties != null) {
+      for (const [key, value] of Object.entries(properties)) {
+        ampIdentify.set(key, value);
+      }
+    }
+    return this.amplitude?.identify(
+      ampIdentify,
+      options,
+    );
+  }
+
+  /**
+   * Set Group for the current user
+   *
+   * @param {String} name
+   * @param {String|String[]} value
+   */
+  setGroup(name, value) {
```

Comment:
Does setGroup allow options input in core browser SDK? If not I think we should create a ticket for that.